### PR TITLE
Fix service port and major refactor

### PR DESCRIPTION
**Pull request by jnsgruk**
This is an unusually big PR that addresses the following issues:

- Remove build artefacts and the built charm file
- Apply automatic formatting with `black`
- Remove unused imports
- Remove the PortainerResources class. In it's previous guise, it wasn't actually *doing anything*
- Simplify the main charm class
- Fix the Kubernetes service definition

If you now build and deploy the charm like so:

```shell
$ charmcraft clean
$ charmcraft pack -v
$ juju deploy ./portainer_ubuntu-20.04-amd64.charm --resource portainer-image=portainer/portainer-ce:latest
```

Once both the containers are running in the pod, you should be able to browse to [http://localhost:30776](http://localhost:30776) and use the UI to create a password and login.

More comments will be added in line to justify some of the changes.

### CODE REVIEW

File: charmcraft.yaml [hunk 1]
```diff
@@ -3,8 +3,13 @@
...
     run-on:
-    - name: "ubuntu"
-      channel: "20.04"
+      - name: "ubuntu"
+        channel: "20.04"
+
+parts:
+  charm:
+    build-packages:
+      - git
```
Comment (jnsgruk) [inline charmcraft.yaml:15]: Added this because we're now pulling the `ops` library from Github

File: src/charm.py [hunk 2]
```diff
@@ -1,177 +1,130 @@
...
+                        kubernetes.client.V1ServicePort(
+                            name="edge",
+                            port=8000,
+                            target_port=8000,
+                            node_port=30777,
+                        ),
+                    ],
+                    selector={
+                        "app.kubernetes.io/name": self.app.name,
+                    },
```
Comment (jnsgruk) [inline src/charm.py:117]: In your previous code, the selector was what prevented the `NodePort` behaving as expected

File: src/charm.py [hunk 3]
```diff
@@ -1,177 +1,130 @@
...
     @property
     def namespace(self) -> str:
+        """Fetch the current Kubernetes namespace by reading it from the service account"""
         with open("/var/run/secrets/kubernetes.io/serviceaccount/namespace", "r") as f:
             return f.read().strip()
 
-    @property
-    def pod_ip(self) -> Optional[IPv4Address]:
-        return IPv4Address(check_output(["unit-get", "private-address"]).decode().strip())
-
```
Comment (jnsgruk) [inline src/charm.py:174]: This was unused

File: src/nginx.yaml [hunk 4]
```diff
@@ -1,23 +0,0 @@
...
-apiVersion: v1
-kind: Service
-metadata:
-  name: nginx-service
-spec:
-    selector:
-      app: nginx
-    ports:
-      - port: 8080
-        targetPort: 80
```
Comment (jnsgruk) [inline src/nginx.yaml:23]: As far as I could tell, this wasn't used anywhere

File: src/resources-nginx.py [hunk 5]
```diff
@@ -1,190 +0,0 @@
-# Copyright 2021 Canonical
```
Comment (jnsgruk) [inline src/resources-nginx.py:1]: As far as I could tell, this wasn't used anywhere

File: src/resources.py [hunk 6]
```diff
@@ -1,130 +0,0 @@
...
-        self.apps_api = kubernetes.client.AppsV1Api(kcl)
-        self.core_api = kubernetes.client.CoreV1Api(kcl)
-        self.auth_api = kubernetes.client.RbacAuthorizationV1Api(kcl)
-
-    def apply(self) -> None:
-        url='https://raw.githubusercontent.com/portainer/k8s/master/deploy/manifests/portainer/portainer.yaml'
-        resp = requests.get(url)
-        logger.info("Creating additional Kubernetes resources")
-        k8s_client=client.api_client.ApiClient(configuration=config.load_incluster_config())
-        ignore_resources = ["Namespace","ServiceAccount","ClusterRoleBinding","PersistentVolumeClaim","Deployment","Service"]
```
Comment (jnsgruk) [inline src/resources.py:35]: This line filtered out *all* of the entries in the file you downloaded, making this whole method (and by extension, class) essentially a no-op

**Review by balasu — APPROVED** [review]
lgtm

File: src/resources.py [hunk 7]
```diff
@@ -1,130 +0,0 @@
...
-        for service in self._services:
-            self.core_api.delete_namespaced_service(
-                namespace=service["namespace"], name=service["body"].metadata.name
-            )
-        # Delete Portainer cluster roles
-        for cr in self._clusterroles:
-            self.auth_api.delete_cluster_role(name=cr["body"].metadata.name)
-        # Delete Portainer cluster role bindings
-        for crb in self._clusterrolebindings:
-            self.auth_api.delete_cluster_role_binding(name=crb["body"].metadata.name)
```
Comment (jnsgruk) [inline src/resources.py:58]: This references properties that don't exist

File: src/resources.py [hunk 8]
```diff
@@ -1,130 +0,0 @@
...
-        return [
-            kubernetes.client.V1VolumeMount(mount_path="/tmp", name="tmp-volume-dashboard"),
-            # kubernetes.client.V1VolumeMount(
-            #     mount_path="/certs", name="portainer-certs"
-            # ),
-            kubernetes.client.V1VolumeMount(
-                mount_path="/var/run/secrets/kubernetes.io/serviceaccount",
-                name="portainer",
-            ),
-        ]
```
Comment (jnsgruk) [inline src/resources.py:88]: This isn't called, and isn't relevant for Portainer

**Pull request merged**